### DISCUSSION TITLE
4-3-1-flight-search Flight search

### DISCUSSION THREAD

**Discussion by aquariusmaster** [bobocode-projects/java-fundamentals-exercises#95] [Q&A]
Цікаво що ви:

- не зробили для FlightService теж інтерфейс (зазвичай ми створюємо для сервису також интерфейс);
- розташували інтерфейс Flights саме у пакеті **service** а не у пакеті **data** (зазвичай ми залишаемо інтерфейси в тому пакеті де імплементація);

Також цікаво, як краще називати інтерфейси та імплементаціі? Наприклад, я з початку переіменував FlightDao в FlightDaoImpl (або SimpleFlightDao) та зробив интерфейс для нього FlightDao (у вас Flights - інтерфейс, FlightDao - імплементація)

**Comment by IgorShandula** [comment]
Також приєднуюсь до цих питань I додам що я б також повернув копію сета з дао findAll() щоб не дозволити редагувати внутрішній стан ззовні, сінглтон, багатопоточну безпеку, але ця вправа не про це мабуть. Можна тести на налл додати І на пошук незалежно від регістру якщо так імплементовано.
